octree/include/pcl/octree/impl/octree_pointcloud.hpp

```diff
@@ -619,7 +659,13 @@ pcl::octree::OctreePointCloud<PointT, LeafContainerT, BranchContainerT, OctreeT>
 {
   OctreeKey key;
 
-  assert(point_idx_arg < input_->size());
+  if (point_idx_arg >= input_->size()) {
+    PCL_ERROR("[pcl::octree::OctreePointCloud::addPointIdx] Input point index (%lu) "
+              "must be < input cloud size (%lu)\n",
+              point_idx_arg,
+              input_->size());
+    return;
+  }
```

Comment:
Should use `assert` here for performance reasons?

Reply:
Yes please